src/engine/Server.cpp

```diff
@@ -696,8 +695,12 @@ boost::asio::awaitable<void> Server::processQuery(
   try {
     auto containsParam = [&params](const std::string& param,
                                    const std::string& expected) {
-      return params.contains(param) && params.at(param).size() == 1 &&
-             params.at(param).front() == expected;
+      auto parameterValue =
+          ad_utility::url_parser::getParameterCheckAtMostOnce(params, param);
+      if (!parameterValue.has_value()) {
+        return false;
+      }
+      return parameterValue.value() == expected;
```

Comment:
```suggestion
     return parameterValue.has_value() && parameterValue.value() == expected;
```
(significantly more concise).